### ISSUE TITLE
Error after unsubscribe - _incomingDataController null

### ISSUE THREAD

**Issue by aaumond** [featurehub-io/featurehub-flutter-sdk#18]
Hi,
I use **featurehub_sse_client** library and I encounter this error:
> [VERBOSE-2:ui_dart_state.cc(209)] Unhandled Exception: Null check operator used on a null value
> #0      EventSource._start.<anonymous closure> (package:featurehub_sse_client/featurehub_sse_client.dart:158:30)
> #1      _rootRunUnary (dart:async/zone.dart:1434:47)
> #2      _CustomZone.runUnary (dart:async/zone.dart:1335:19)
> #3      _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
> #4      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
> #5      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
> #6      _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:123:11)
> #7      _HandleErrorStream._handleData (dart:async/stream_pipe.dart:253:10)

The error occurs after last listener stop to listen event source but incoming data continue to be sent by back-end.

In library, the error occured cause of `_incomingDataController` (set to `null` by `_lastStreamDisconnected` function).
I think listening on the response stream (`response.stream.listen`) should be stopped when the last listening on the event source is canceled. 
```
response.stream.listen((value) {
      _incomingDataController!.add(value);
    },
```

I remain available for further information if needed.
Thank you in advance.

**Comment by aaumond** [comment]
Hello,
Do you have any idea when the fix will be added?

**Comment by rvowles** [comment]
This library exists purely to support our FeatureHub SDK, we have a long list of other things we are dealing with and unless it affects that SDK it has a very low priority i'm sorry. If you come up with a fix and an associated test I'm happy to add it.

**Comment by aaumond** [comment]
@rvowles Thank you for your reply.
The fix I suggest that works for me is as follows:

Add a private field on `EventSource` class :
``StreamSubscription? _subscription;``

Then, add an affectation like this when you listen response stream in `_start` function:
``_subscription = response.stream.listen((value)``

And finally, add this instruction in `_lastStreamDisconnected`  after `await _incomingDataController!.close();`  instruction:
``await _subscription?.cancel();``

I remain available if you have any questions.
Thanks in advance.
Anthony

**Comment by aaumond** [comment]
@rvowles Hello, were you able to see the fix I described?
Thank you in advance.

**Comment by rvowles** [comment]
@aaumond could you have a look at the attached PR please?

**Comment by aaumond** [comment]
@rvowles I tested the patch. It works well. The PR is ok for me

**Comment by rvowles** [comment]
this was released as 1.3.2
